design_spec.md

```diff
@@ -0,0 +1,205 @@
+# Design Specification Document: Gazebo Documentation Index
+
+## 1. Introduction
+
+The Gazebo project has a vast set of learning resources in the form of a
+documentation section, Gazebo tutorials, the Q&A website, ROS answers and other
+blogs that developers can refer to for any assistance. All of this information
+is distributed across the internet with some links joining each other. 
+
+The aim of this project is to bring all the learning material under one webpage
+in the form of a documentation index that contains links to the content where
+the respective information is hosted. 
+
+## 2. Background
+At present, the learning resources for Gazebo are distributed over the internet
+in the form of Gazebo’s official documentation, the Q&A website and ROS-Answers.
+Some noteworthy help can be found through examples and explanations in the
+comments of Bitbucket issues and Gazebo’s API. There are also third-party
+sources that provide video tutorials and blog posts that are helpful in
+learning Gazebo. It can be a bit overwhelming to keep track of one’s learning
+when the content is distributed as in this case.
+
+A documentation index is a platform where links to relevant learning resources
+for a software system are indexed to allow users to find any help at one place.
+
+#### Some common documentation indexes by - 
+
+- [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Index)
+- [Intel](https://www.intel.com/content/www/us/en/programmable/documentation/lit-index.html)
+- [Oracle](https://www.oracle.com/technetwork/indexes/documentation/index-100966.html)
+
+## 3. Purpose
+
+A documentation index is a neat way to accommodate links to all the relevant
+learning content into one webpage. It is very convenient since almost any help
+is just a page quicksearch away. The user can think of related keywords or
+categories and then look through the index to find the relevant information.
+Such a platform can act as a one-stop place to get all pertinent information about Gazebo.
+
+
+1. Compilation of best learning resources from across the internet, including 
+tutorials, third-party blog posts, Github issues' comments and the Gazebo answers website,
+
+2. All relevant content under one roof.
+
+## 4. Implementation
+
+### 4.1 Technology stack:
+
+[Jekyll](https://jekyllrb.com/): 
+- Static site generation
+- Easy maintenance and collaboration with yml-based data for index items
+- Reusable HTML templates
+
+[Travis CI](https://travis-ci.org/): 
+- Index integrity tests
+- Continuous integration
+
+[AWS](https://aws.amazon.com/): 
+- Deployment and hosting
+
+### 4.2 Index structure
+
+The documentation index items have been divided by categories and subcategories.
```

Comment:
For me is clear that the maximum deep level for the index is two (categories >> subcategories) but I would like to see it written explicitly somewhere in these sentences.

Reply:
Okay, I will add that.